course_catalog/etl/loaders.py

```diff
@@ -209,7 +223,26 @@ def load_program(program_data, blacklist):
 def load_programs(programs_data):
     """Load a list of programs"""
     blacklist = load_course_blacklist()
-    return [load_program(program_data, blacklist) for program_data in programs_data]
+
+    if (
+        programs_data
+        and len(programs_data) > 0
+        and programs_data[0].get("courses")
+        and len(programs_data[0].get("courses")) > 0
+    ):
+        platform = programs_data[0].get("courses")[0].get("platform")
```

Comment:
I think you can make this more straightforward by having `platform` be passed as a keyword argument to a [curried version](https://toolz.readthedocs.io/en/latest/api.html#toolz.functoolz.curry) of `load_programs`, which would be paramterized in `pipelines.py`.